presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -231,6 +274,36 @@ public int putIfAbsent(int position, Page page)
         return putIfAbsent(position, page, rawHash);
     }
 
+    @Override
+    public boolean indeterminateContains(int position, Page page, int[] hashChannels)
+    {
+        if (!equalFunctions.isPresent()) {
+            return false;
+        }
+        // indeterminateContains only support 1 column and the only column must be one of ArrayType, MapType, RowType
+        checkState(hashChannels.length == 1, "the size of hashChannels must be 1");
+        if (equalFunctions.isPresent()) {
+            MethodHandle function = equalFunctions.get().get(0);
+            Object currentValue = readNativeValue(hashTypes.get(hashChannels[0]), page.getBlock(0), position);
+            for (int i = 0; i < nullRows.size(); i++) {
+                try {
+                    Boolean result = (Boolean) function.invoke(currentValue, nullRows.get(i).get(0));
```

Comment:
If values in values in hash are all determinated, is it possible that result of equality with any of them against `currentValue` returns null?

It sounds strange to me. If not, then I would only update `ChannelSet#containsNull` to true if any of hash values is indeterminate.

Moreover I would prefer not to call scalar function here, please use `ProjectNode` for that.